lib/neo4j/active_node/has_n.rb

```diff
@@ -516,9 +522,13 @@ def build_association(macro, direction, name, options)
       end
 
       def add_association(name, association_object)
-        @associations ||= {}
-        fail "Association `#{name}` defined for a second time.  Associations can only be defined once" if @associations.key?(name)
-        @associations[name] = association_object
+        fail "Association `#{name}` defined for a second time. "\
+             'Associations can only be defined once' if duplicate_association?(name)
+        associations[name] = association_object
+      end
+
+      def duplicate_association?(name)
+        associations.key?(name) && parent_associations[name] != associations[name]
```

Comment:
I wonder if you're fixing the problem that I mentioned above here (maybe you already knew that ;).  Either way, nice refactor